# HG changeset patch
# User Rik <rik@octave.org>
# Date 1437506496 25200
#      Tue Jul 21 12:21:36 2015 -0700
# Node ID 07ddaaeb90407fa32f82413a7b23b270c63e1044
# Parent  f74ab65ee1bf33383e73f05cbbc780a3afc11a89
Remove all temporary documentation files with 'maintainer-clean' target (bug #45574).

* doc/module.mk: Remove --build-dir option from octave.dvi and octave.pdf rules.
Add rules to doc-clean target to remove cache directiories created by texi2dvi
and texi2pdf.

* acinclude.m4: Use a different temporary build-dir for TEXI2DVI and TEXI2PDF
so that they can run in parallel without colliding.

diff --git a/doc/module.mk b/doc/module.mk
--- a/doc/module.mk
+++ b/doc/module.mk
@@ -247,23 +247,23 @@ doc/interpreter/octave.pdf: $(DOC_IMAGES
 	  $(am__cd) $(srcdir) && \
 	  $$restore $$backupdir/* `echo "./$@" | sed 's|[^/]*$$||'`; \
 	fi; \
 	rm -rf $$backupdir; exit $$rc
 
 doc/interpreter/octave.dvi: doc/interpreter/octave.texi $(srcdir)/doc/interpreter/version-octave.texi | doc/interpreter/$(am__dirstamp)
 	$(AM_V_TEXI2DVI)TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
 	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I doc/interpreter -I $(abs_top_srcdir)/doc/interpreter' \
-	$(TEXI2DVI) $(AM_V_texinfo) --build-dir=$(@:.dvi=.t2d) -o $@ $(AM_V_texidevnull) \
+	$(TEXI2DVI) $(AM_V_texinfo) -o $@ $(AM_V_texidevnull) \
 	`test -f 'doc/interpreter/octave.texi' || echo '$(abs_top_srcdir)/'`doc/interpreter/octave.texi
 
 doc/interpreter/octave.pdf: doc/interpreter/octave.texi $(srcdir)/doc/interpreter/version-octave.texi | doc/interpreter/$(am__dirstamp)
 	$(AM_V_TEXI2PDF)TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
 	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I doc/interpreter -I $(abs_top_srcdir)/doc/interpreter' \
-	$(TEXI2PDF) $(AM_V_texinfo) --build-dir=$(@:.pdf=.t2p) -o $@ $(AM_V_texidevnull) \
+	$(TEXI2PDF) $(AM_V_texinfo) -o $@ $(AM_V_texidevnull) \
 	`test -f 'doc/interpreter/octave.texi' || echo '$(abs_top_srcdir)/'`doc/interpreter/octave.texi
 
 $(OCTAVE_HTML_STAMP): doc/interpreter/octave.texi $(srcdir)/doc/interpreter/version-octave.texi | doc/interpreter/$(am__dirstamp)
 	$(AM_V_MAKEINFO)rm -rf $(OCTAVE_HTML_DIR)
 	$(AM_V_at)if $(MAKEINFOHTML) $(AM_MAKEINFOHTMLFLAGS) $(MAKEINFOFLAGS) -I doc/interpreter -I $(abs_top_srcdir)/doc/interpreter \
 	 -o $(OCTAVE_HTML_TMP_DIR) `test -f 'doc/interpreter/octave.texi' || echo '$(abs_top_srcdir)/'`doc/interpreter/octave.texi; \
 	then \
 	  rm -rf $(OCTAVE_HTML_DIR)@ && \
@@ -290,21 +290,16 @@ doc_EXTRA_DIST += \
   doc/interpreter/TODO \
   doc/interpreter/doc-cache \
   doc/interpreter/octave.dvi \
   doc/interpreter/octave.ps \
   doc/interpreter/octave.pdf \
   doc/interpreter/octave.html \
   $(HTMLDIR_IMAGES)
 
-## The texi2dvi script (used to create both PDF and DVI output formats)
-## uses some fixed temporary file names.  In order to avoid a race condition
-## the DVI and PDF builds are forced to run serially through a Makefile rule.
-#doc/interpreter/octave.pdf: doc/interpreter/octave.dvi
-
 # Prevent packaging of distribution unless all libraries
 # necessary to create documentation are present
 doc-interpreter-dist-hook:
 	@$(GREP) '#define HAVE_COLAMD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing COLAMD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_CHOLMOD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing CHOLMOD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_UMFPACK 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing UMFPACK library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_QHULL 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing QHULL library.  Cannot package distribution!" ; exit 1; }
 
@@ -536,14 +531,16 @@ spellcheck: $(SPELLCHECK_FILES)
 EXTRA_DIST += $(doc_EXTRA_DIST)
 
 CLEANFILES += $(doc_CLEANFILES)
 DISTCLEANFILES += $(doc_DISTCLEANFILES)
 MAINTAINERCLEANFILES += $(doc_MAINTAINERCLEANFILES)
 
 doc-clean:
 	rm -f $(doc_CLEANFILES)
+	rm -rf $(top_builddir)/.t2d.cache
+	rm -rf $(top_builddir)/.t2p.cache
 
 doc-distclean: doc-clean
 	rm -f $(doc_DISTCLEANFILES)
 
 doc-maintainer-clean: doc-distclean
 	rm -f $(doc_MAINTAINERCLEANFILES)
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2269,17 +2269,21 @@ AC_DEFUN([OCTAVE_PROG_TEXI2DVI], [
     TEXI2DVI='$(top_srcdir)/build-aux/missing texi2dvi'
     warn_texi2dvi="
 
 I didn't find texi2dvi, but it's only a problem if you need to
 reconstruct the DVI version of the manual
 "
     OCTAVE_CONFIGURE_WARNING([warn_texi2dvi])
   fi
-  TEXI2DVI="$TEXI2DVI"' --build-dir=t2d_cache'
+dnl FIXME: texi2pdf and texi2dvi can not be run simultaneously as they both
+dnl        attempt to create a temporary build directory of the same name.
+dnl        A workaround is to define separate build directories for each
+dnl        command as is done here.
+  TEXI2DVI="$TEXI2DVI"' --build-dir=.t2d.cache'
   AC_SUBST(TEXI2DVI)
 ])
 dnl
 dnl Check for texi2pdf.
 dnl
 AC_DEFUN([OCTAVE_PROG_TEXI2PDF], [
   AC_REQUIRE([OCTAVE_PROG_TEXI2DVI])
   AC_CHECK_PROG(TEXI2PDF, texi2pdf, texi2pdf, [])
@@ -2296,17 +2300,21 @@ AC_DEFUN([OCTAVE_PROG_TEXI2PDF], [
     TEXI2PDF='$(top_srcdir)/build-aux/missing texi2pdf'
     warn_texi2pdf="
 
 I didn't find texi2pdf, but it's only a problem if you need to
 reconstruct the PDF version of the manual
 "
     OCTAVE_CONFIGURE_WARNING([warn_texi2pdf])
   fi
-  TEXI2PDF="$TEXI2PDF"' --build-dir=t2d_cache'
+dnl FIXME: texi2pdf and texi2dvi can not be run simultaneously as they both
+dnl        attempt to create a temporary build directory of the same name.
+dnl        A workaround is to define separate build directories for each
+dnl        command as is done here.
+  TEXI2PDF="$TEXI2PDF"' --build-dir=.t2p.cache'
   AC_SUBST(TEXI2PDF)
 ])
 dnl
 dnl Set default value for a variable and substitute it.
 dnl
 AC_DEFUN([OCTAVE_SET_DEFAULT], [
   ifelse($#, 2, [: ${$1=$2}
 ])dnl
